polkadot/runtime/common/src/traits.rs

```diff
@@ -254,12 +254,3 @@ pub trait Auctioneer<BlockNumber> {
 	/// Check if the para and user combination has won an auction in the past.
 	fn has_won_an_auction(para: ParaId, bidder: &Self::AccountId) -> bool;
 }
-
-/// Runtime hook for when we swap a lease holding parachain and an on-demand parachain.
-#[impl_trait_for_tuples::impl_for_tuples(30)]
-pub trait OnSwap {
```

Comment:
I don't think moving this is necessary. The wiring has to happen at the runtime level anyways, we can also implement the trait there and just call into the coretime pallet.

Reply:
Yes, that's a good idea.